RELEASE.md

```diff
@@ -1,20 +1,17 @@
 # Cutting a Jaeger UI release
 
-1. Determine the version for the release.
```

Comment:
I felt the instructions could be improved to be a bit more succinct.

Reply:
I don't know, don't think original text hurts, especially if a new person is doing it for the first time

Reply:
but I see you copied it below, so good